src/http.js

```diff
@@ -170,4 +171,13 @@ export default class HTTP {
     }
     this.events.emit("backoff", backoffMs);
   }
+
+  _checkForRetryAfterHeader(status, headers) {
+    let retryAfter = headers.get("Retry-After");
+    if (!retryAfter) {
+      return;
+    }
+    retryAfter = parseInt(retryAfter, 10);
+    this.events.emit("retry-after", retryAfter);
```

Comment:
Nit: I wonder if we shouldn't explicitly mention the unit in the event name, eg. `retry-after-seconds`; yeah it's ugly but in my experience it prevents a lot of mistakes.